Cannot build from source nor build from AUR

Hello everyone!
I am trying to install the game on my computer. so I tried installing from the AUR and it failed it gave me an error message similar to the same one here https://github.com/colobot/colobot/issues/1180.
After that I have tried to install from source which also failed(both stable branch and dev branch), I even ran cmake with ignore errors flag which did let it compile but the game couldn't launch.

```
/tmp/pamac-build-oren/colobot-gold/src/colobot-colobot-gold-0.1.11.1-alpha/src/graphics/model/model_output.cpp: In function ‘void Gfx::ModelOutput::WriteOldModel(const Gfx::CModel&, std::ostream&)’:
/tmp/pamac-build-oren/colobot-gold/src/colobot-colobot-gold-0.1.11.1-alpha/src/graphics/model/model_output.cpp:331:16: error: ‘char* strncpy(char*, const char*, size_t)’ specified bound 20 equals destination size [-Werror=stringop-truncation]
         strncpy(t.texName, triangle.tex1Name.c_str(), 20);
         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

cc1plus: all warnings being treated as errors
make[2]: *** [src/CMakeFiles/colobotbase.dir/build.make:752: src/CMakeFiles/colobotbase.dir/graphics/model/model_output.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:187: src/CMakeFiles/colobotbase.dir/all] Error 2
```
I also tried to change this part to `strncpy(t.texName, triangle.tex1Name.c_str(), 20);` writing 19 instead of 20 but then I ran to another error message.

I have cmake 3.11.4-2
gcc 8.2.0-2

when compiling the game with errors:
```
Error in CEngine::Init()

Please see the console output or log file
to get more information on the source of error


```
what should I do ?
thanks for the help

--- a/src/graphics/model/model_io_structs.h
+++ b/src/graphics/model/model_io_structs.h
@@ -188,7 +188,7 @@ struct OldModelTriangleV1
     Vertex p2;
     Vertex p3;
     Material material;
-    char texName[20] = {};
+    char texName[21] = {'\0'};
     float min = 0;
     float max = 0;
 };
@@ -207,7 +207,7 @@ struct OldModelTriangleV2
     Vertex p2;
     Vertex p3;
     Material material;
-    char texName[20] = {};
+    char texName[21] = {'\0'};
     float min = 0.0f;
     float max = 0.0f;
     long state = 0;
@@ -231,7 +231,7 @@ struct OldModelTriangleV3
     VertexTex2 p2;
     VertexTex2 p3;
     Material   material;
-    char texName[20] = {};
+    char texName[21] = {'\0'};
     float min = 0.0f;
     float max = 0.0f;
     long state = 0;

